### PR TITLE
Recommended channels

### PR DESCRIPTION
**Pull request by norkans7**
Only merge after #1452

### CODE REVIEW

**Comment by rowanseymour** [comment]
@nicpottier yeah the only place we should be resorting to literal channel codes is in the handlers and handler tests - since that code is going to be replaced. Elsewhere it's probably a sign that some functionality belongs in the channel type package.

**Comment by norkans7** [comment]
I thought we can remove those connect links from the claim page however those are providing a better UX when the user is not sure why he can't connect Twilio or Nexmo.
I can use pre_process on the dynamic types claim view to redirect to the org connect view if the org in not connected to the aggregator if we agree to not use the modal to add the Twilio/Nexmo account we'll redirect to a new page entirely. That way no channel type code in the template.

File: templates/channels/channel_claim.haml [hunk 1]
```diff
@@ -17,40 +17,68 @@
...
 
     -if not user_org.is_connected_to_twilio
-      .claim-row.twilio-channel{class:"{% if recommended_channel == 'T' %}recommended{% endif %}"}
-        .recommendation.hide
-          %h3
-            -trans "Recommendation"
-          -blocktrans with "<a href='http://twilio.com'>" as link_start and "</a>" as link_end
-            We recommend using a Twilio phone number for SMS in your country. Sign up for a {{link_start}}Twilio Account{{link_end}} if you don't already have one.
-
+      .claim-row.twilio-channel{class:"{% if 'T' in recommended_channels_to_connect %}recommended{% endif %}"}
```
[the remark this answers is not in the document]
Comment (norkans7) [reply templates/channels/channel_claim.haml]: That's not need we can just use pre_process and redirect to the connect view as long as we agree we won't use the pop modal as we used to

[the remark this answers is not in the document]
Comment (rowanseymour) [reply templates/channels/channel_claim.haml]: you mean if/when they try to claim a Twilio/Nexmo number, we will redirect them to the Twilio/Nexmo account page first if they don't already have an account configured at the org level?

[the remark this answers is not in the document]
Comment (norkans7) [reply templates/channels/channel_claim.haml]: Yep

[the remark this answers is not in the document]
Comment (norkans7) [reply templates/channels/channel_claim.haml]: That's exactly what these block here are for

[the remark this answers is not in the document]
Comment (rowanseymour) [reply templates/channels/channel_claim.haml]: I'm not quite following - I was thinking these blocks aren't needed if you are gonna redirect attempts to claim Twilio and Nexmo channels when they don't have credentials stored in the org, i.e. 

- if Twilio and Nexmo channels are available in the user's country, we show the claim blocks regardless of whether they have setup an org level account.
- when they go to those claim views, we redirect if they first need to setup org-level credentials

[the remark this answers is not in the document]
Comment (norkans7) [reply templates/channels/channel_claim.haml]: Right, currently we do not really redirect to the org connect views,
we use JS to show a modal to the user and use formax to POST to the connect view.

Now if wee agree to use redirect we can remove there JS modal here and there block with channel_type constants will not be needed as well

**Comment by nicpottier** [comment]
Wouldn't it be cleaner to use `settings.CHANNEL_TYPES` for ordering?

**Comment by nicpottier** [comment]
Maybe by using an ordered_dict in `channels/__init__.py`?

**Comment by norkans7** [comment]
the ordering is from https://github.com/nyaruka/rapidpro/blob/603c35bea6c33ba20f8740b6ff0545270d8f7276/temba/channels/views.py#L1408

File: temba/channels/models.py [hunk 2]
```diff
@@ -98,12 +98,20 @@ class IVRProtocol(Enum):
 
     ivr_protocol = None
 
+    ordering = 20
```
Comment (rowanseymour) [inline temba/channels/models.py]: Can we use the order in settings.CHANNEL_TYPES for this? That seems easier than channel types declaring their own ordering.

Comment (rowanseymour) [reply temba/channels/models.py]: I know it's been A-Z up until now but we can re-organize it - break them up by categories etc

Comment (rowanseymour) [reply temba/channels/models.py]: Oh just saw @nicpottier's comments saying same thing

Comment (norkans7) [reply temba/channels/models.py]: Yep, I change to that then

Comment (norkans7) [reply temba/channels/models.py]: I think having a field on ChannelType is easier, we use an iterator on Channel.get_types and in .types we use a dict to store those I keep the ordering in there then

**Comment by norkans7** [comment]
I used settings.CHANNEL_TYPES order to control the ordering on claim page and remove the use of channel_type code in the template, now we redirect to the org connect views in TWILIO,NEXMO or PLIVO are not connected to the org

File: temba/channels/types/dartmedia/type.py [hunk 3]
```diff
@@ -40,6 +40,9 @@ def is_available_to(self, user):
         org = user.get_org()
         return org.timezone and six.text_type(org.timezone) in ["Asia/Jakarta"]
 
+    def is_recommended_to(self, user):
+        self.is_available_to(user)
```
Comment (rowanseymour) [inline temba/channels/types/dartmedia/type.py]: *return

**Comment by nicpottier** [comment]
NIce lets merge all this stuff and ship to staging again.

File: temba/channels/types/blackmyna/type.py [hunk 4]
```diff
@@ -34,6 +34,9 @@ def is_available_to(self, user):
         org = user.get_org()
         return org.timezone and six.text_type(org.timezone) in ["Asia/Kathmandu"]
 
+    def is_recommended_to(self, user):
+        self.is_available_to(user)
```
Comment (rowanseymour) [inline temba/channels/types/blackmyna/type.py]: *return

File: temba/channels/types/globe/type.py [hunk 5]
```diff
@@ -36,6 +36,9 @@ def is_available_to(self, user):
         org = user.get_org()
         return org.timezone and six.text_type(org.timezone) in ['Asia/Manila']
 
+    def is_recommended_to(self, user):
+        return self.is_available_to(user)
```
Comment (rowanseymour) [inline temba/channels/types/globe/type.py]: *return

**Pull request merged**